### PR TITLE
Reduce runtime of h2o-algos JUnits by running 5 H2O clusters in parallel. Also run them in single-node mode.

### PR DESCRIPTION
**Pull request by arnocandel**
#
This change is [Reviewable](https://reviewable.io/reviews/h2oai/h2o-3/189)

**Before:** All h2o-algos JUnits run in distributed mode.  There's one 5-node H2O cluster.  Took about 40 minutes on a dual-Xeon server.

11:14:06 Running h2o-algos junit tests...
11:53:15 h2o-algos junit tests PASSED
11:53:15 :h2o-algos:testMultiNode took 2349.323 secs

**Now:** All h2o-algos JUnits run first in single-node mode, and if that passes, then they all run again in distributed mode (4 JVMs). For both cases, there's 5 H2O clusters in parallel.  Takes less than 15 minutes for both.

16:58:16 :h2o-algos:testSingleNode
16:58:16 Running h2o-algos junit tests...
17:01:29 h2o-algos junit tests PASSED
17:01:29 :h2o-algos:testSingleNode took 192.303 secs
17:01:29 :h2o-algos:testMultiNode
17:01:29 Running h2o-algos junit tests...
17:16:06 h2o-algos junit tests PASSED
17:16:06 :h2o-algos:testMultiNode took 877.816 secs

http://172.17.2.151:8080/view/arno/job/arno_junit/445/console

Side-effect of this is that the memory requirements went from 5x3GB to 20x2.5GB, but that should be fine for servers.

Once we create a separate suite for distributed JUnits (hopefully fewer tests), we should be able to reduce the memory requirements again.

### CODE REVIEW

File: h2o-algos/testMultiNode.sh [hunk 1]
```diff
@@ -21,11 +21,11 @@ case "`uname`" in
...
 
 function cleanup () {
-  kill -9 ${PID_1} ${PID_2} ${PID_3} ${PID_4} 1> /dev/null 2>&1
+  kill -9 ${PID_11} ${PID_21} ${PID_31} ${PID_41} ${PID_51} 1> /dev/null 2>&1
   wait 1> /dev/null 2>&1
-  RC=`cat $OUTDIR/status.0`
-  if [ $RC -ne 0 ]; then
-    cat $OUTDIR/out.0
+  RC="`paste $OUTDIR/status.* | sed 's/\s*//g'`"
+  if [ $RC != "00000" ]; then
```
Comment (mmalohlava) [inline h2o-algos/testMultiNode.sh]: Better is "$RC" != "00000"

Comment (mmalohlava) [inline h2o-algos/testMultiNode.sh]: I am not sure if `exit $RC` will work properly with `exit 00000` on MacOSX.

From: http://tldp.org/LDP/abs/html/exit-status.html

> Every command returns an exit status (sometimes referred to as a return status or exit code). A successful command returns a 0, while an unsuccessful one returns a non-zero value that usually can be interpreted as an error code. Well-behaved UNIX commands, programs, and utilities return a 0 exit code upon successful completion, though there are some exceptions.

File: h2o-algos/testMultiNode.sh [hunk 2]
```diff
@@ -116,8 +117,13 @@ fi
 # Launch last driver JVM.  All output redir'd at the OS level to sandbox files.
 echo Running h2o-algos junit tests...
 
-($JVM -Ddoonly.tests=$DOONLY -Dbuild.id=$BUILD_ID -Dignore.tests=$IGNORE -Djob.name=$JOB_NAME -Dgit.commit=$GIT_COMMIT -Dgit.branch=$GIT_BRANCH -Dai.h2o.name=$CLUSTER_NAME -Dai.h2o.baseport=$CLUSTER_BASEPORT -Dai.h2o.ga_opt_out=yes $JACOCO_FLAG $JUNIT_RUNNER $JUNIT_TESTS_BOOT `cat $OUTDIR/tests.txt` 2>&1 ; echo $? > $OUTDIR/status.0) 1> $OUTDIR/out.0 2>&1
+($JVM -Ddoonly.tests=$DOONLY -Dbuild.id=$BUILD_ID -Dignore.tests=$IGNORE -Djob.name=$JOB_NAME -Dgit.commit=$GIT_COMMIT -Dgit.branch=$GIT_BRANCH -Dai.h2o.name=$CLUSTER_NAME.1 -Dai.h2o.baseport=$CLUSTER_BASEPORT -Dai.h2o.ga_opt_out=yes $JACOCO_FLAG $JUNIT_RUNNER $JUNIT_TESTS_BOOT `cat $OUTDIR/tests.txt | sed -n '1~5p'` 2>&1 ; echo $? > $OUTDIR/status.1) 1> $OUTDIR/out.1 2>&1 & PID_1=$!
```
Comment (mmalohlava) [inline h2o-algos/testMultiNode.sh]: `sed -n '1~5p'` does not work on Mac OS X

**Comment by arnocandel** [comment]
Trying to reduce the memory usage to 2GB, and going back up to 4 nodes.

**Comment by arnocandel** [comment]
Testing on arno_jenkins, will merge once it passes 5 times in a row.

**Pull request merged**